community/kernel/src/main/java/org/neo4j/unsafe/impl/batchimport/store/BatchingIdSequence.java

```diff
@@ -32,11 +32,8 @@
     @Override
     public long nextId()
     {
-        long result = nextId++;
-        if ( result == IdGeneratorImpl.INTEGER_MINUS_ONE )
-        {
-            result = nextId++;
-        }
+        long result = peek();
```

Comment:
This bit here can arguably be left out (to make least possible change), but I consider it incorrect that `peek` might return anything else than `nextId` will.